Model/Logger.php

```diff
@@ -48,19 +48,21 @@ public function __construct(
      * @param string|null $rvvupOrderId
      * @param string|null $rvvupPaymentId
      * @param string|null $magentoOrderId
+     * @param string|null $origin
      * @return bool
      */
     public function addRvvupError(
         string $message,
         ?string $cause = null,
         ?string $rvvupOrderId = null,
         ?string $rvvupPaymentId = null,
-        ?string $magentoOrderId = null
+        ?string $magentoOrderId = null,
+        ?string $origin = null
     ) {
         $result = $this->addRecord(
             static::ERROR,
             $message,
-            [$cause, $rvvupOrderId, $rvvupPaymentId, $magentoOrderId]
+            [$cause, $rvvupOrderId, $rvvupPaymentId, $magentoOrderId, $origin]
         );
 
         try {
```

Comment:
$data = $this->prepareData($message, $cause, $rvvupOrderId, $rvvupPaymentId, $magentoOrderId);

should also include $origin, as its not writing that currently.